internal/pkg/plugininstaller/common/repo.go

```diff
@@ -30,54 +31,36 @@ type Repo struct {
 
 // BuildRepoRenderConfig will generate template render variables
 func (d *Repo) BuildRepoRenderConfig() map[string]interface{} {
+	repoInfo := d.BuildRepoInfo()
 	renderConfigMap := map[string]interface{}{
 		"AppName": d.Repo,
 		"Repo": map[string]string{
 			"Name":  d.Repo,
-			"Owner": d.getRepoOwner(),
+			"Owner": repoInfo.GetRepoOwner(),
 		},
 	}
 	return renderConfigMap
 }
 
 // CreateGithubClient build github client connection info
-func (d *Repo) CreateGithubClient(needAuth bool) (*github.Client, error) {
-	ghOptions := &github.Option{
-		Owner:    d.Owner,
-		Org:      d.Org,
-		Repo:     d.Repo,
-		NeedAuth: needAuth,
-	}
-	ghClient, err := github.NewClient(ghOptions)
-	if err != nil {
-		return nil, err
-	}
-	return ghClient, nil
-}
-
-// CreateGitlabClient build gitlab connection info
-func (d *Repo) CreateGitlabClient() (*gitlab.Client, error) {
-	return gitlab.NewClient(gitlab.WithBaseURL(d.BaseURL))
-}
-
-// buildgitlabOpts build gitlab connection options
-func (d *Repo) buildgitlabOpts() *gitlab.CreateProjectOptions {
-	return &gitlab.CreateProjectOptions{
-		Name:       d.Repo,
-		Branch:     d.getBranch(),
-		Namespace:  d.Org,
-		Visibility: d.Visibility,
+func (d *Repo) NewClient() (repo.ClientOperation, error) {
```

Comment:
The original code get client supports GitHub and GitLab, why does it only support GitHub here?